### PR TITLE
EditorFeatures.Cocoa: Bump macos workload to RC3

### PR DESCRIPTION
**Pull request by sandyarmstrong**

### CODE REVIEW

**Comment by sandyarmstrong** [comment]
This probably requires a corresponding change in the editor to resolve inconsistencies caused by the workload assembly changing from Xamarin.Mac.dll to Microsoft.macOS.dll.

**Comment by sandyarmstrong** [comment]
No idea what this is about:

> ##[error]src\VisualStudio\VisualBasic\Impl\LanguageService\VisualBasicPackage.vb(60,29): error BC30456: (NETCORE_ENGINEERING_TELEMETRY=Build) 'ComAggregate' is not a member of 'Microsoft.VisualStudio.Interop'.

**Comment by sandyarmstrong** [comment]
@sharwell @genlu any idea about this `'ComAggregate' is not a member of 'Microsoft.VisualStudio.Interop'` error? I don't think I can avoid bumping editor/shell/etc to 17.3.

File: eng/Versions.props [hunk 1]
```diff
@@ -25,20 +25,20 @@
...
     <CodeStyleAnalyzerVersion>4.2.0-2.final</CodeStyleAnalyzerVersion>
-    <VisualStudioEditorPackagesVersion>17.2.3194</VisualStudioEditorPackagesVersion>
+    <VisualStudioEditorPackagesVersion>17.3.67-preview-ga1f07c2812</VisualStudioEditorPackagesVersion>
     <ILAsmPackageVersion>5.0.0-alpha1.19409.1</ILAsmPackageVersion>
     <ILDAsmPackageVersion>5.0.0-preview.1.20112.8</ILDAsmPackageVersion>
     <MicrosoftVisualStudioLanguageServerProtocolPackagesVersion>17.3.5</MicrosoftVisualStudioLanguageServerProtocolPackagesVersion>
-    <MicrosoftVisualStudioShellPackagesVersion>17.2.32505.113</MicrosoftVisualStudioShellPackagesVersion>
-    <RefOnlyMicrosoftBuildPackagesVersion>16.5.0</RefOnlyMicrosoftBuildPackagesVersion>
+    <MicrosoftVisualStudioShellPackagesVersion>17.3.0-preview-1-32407-044</MicrosoftVisualStudioShellPackagesVersion>
+    <RefOnlyMicrosoftBuildPackagesVersion>17.1.0</RefOnlyMicrosoftBuildPackagesVersion>
```
Comment (sharwell) [inline eng/Versions.props:33]: This is intentionally locked to 16.5. See #61265

```suggestion
    <RefOnlyMicrosoftBuildPackagesVersion>16.5.0</RefOnlyMicrosoftBuildPackagesVersion>
```

Comment (sandyarmstrong) [reply eng/Versions.props]: This alone wasn't enough to fix the build locally. I'm just going to undo all these version changes and only bump the Mac-specific editor packages.

File: eng/Versions.props [hunk 2]
```diff
@@ -26,6 +26,7 @@
     <!-- CodeStyleAnalyzerVersion should we updated together with version of dotnet-format in dotnet-tools.json -->
     <CodeStyleAnalyzerVersion>4.2.0-2.final</CodeStyleAnalyzerVersion>
     <VisualStudioEditorPackagesVersion>17.2.3194</VisualStudioEditorPackagesVersion>
+    <VisualStudioMacEditorPackagesVersion>17.3.67-preview-ga1f07c2812</VisualStudioMacEditorPackagesVersion>
```
Comment (sandyarmstrong) [inline eng/Versions.props:29]: We can change this to `$(VisualStudioEditorPackagesVersion)` after my editor changes have merged in VS and Roslyn is able to take those newer packages and dependencies.

**Comment by sandyarmstrong** [comment]
I notice the merge button is enabled even though Build Analysis and roslyn-integration-corehost are failing (I tried rerunning them). Is it OK if I merge?

**Comment by davidwengier** [comment]
Yes, corehost tests aren't blocking yet, as they're not stavle.

Either way "but GitHub let me" is a totally valid excuse if you're not an admin 😁

File: src/EditorFeatures/Core.Cocoa/Microsoft.CodeAnalysis.EditorFeatures.Cocoa.csproj [hunk 3]
```diff
@@ -45,14 +45,15 @@
...
+    <PackageReference Include="System.Runtime.InteropServices.NFloat.Internal" Version="$(SystemRuntimeInteropServicesNFloatInternalVersion)" />
     <PackageReference Include="System.ValueTuple" Version="$(SystemValueTupleVersion)" />
 
     <!-- This project should target net6.0-macos, but due to repeated issues
          provisioning the macos workload on Windows (generally hitting signature
          validation errors on workload dependencies), we are instead manually
          referencing what we need out of the macos SDK. -->
     <PackageDownload Include="Microsoft.macOS.Ref" Version="[$(MicrosoftmacOSRefVersion)]" />
-    <Reference Include="$(NuGetPackageRoot)microsoft.macos.ref\$(MicrosoftmacOSRefVersion)\ref\net6.0\Xamarin.Mac.dll" />
+    <Reference Include="$(NuGetPackageRoot)microsoft.macos.ref\$(MicrosoftmacOSRefVersion)\ref\net6.0\Microsoft.macOS.dll" />
```
Comment (davidwengier) [inline src/EditorFeatures/Core.Cocoa/Microsoft.CodeAnalysis.EditorFeatures.Cocoa.csproj:56]: @sandyarmstrong this is the same change as you're making in https://github.com/dotnet/roslyn/pull/61250

Is this a problem? Or do we need to wait for the 6.0.300 bump in VS Mac?

**Pull request merged**